File: tests/fixtures/real_world/python/state/socket_lifecycle.py
import socket

def connect_and_leak(host, port):
    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    s.connect((host, port))
    s.send(b'hello')
    data = s.recv(1024)
    return data

def connect_and_close(host, port):
    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    s.connect((host, port))
    try:
        s.send(b'hello')
        data = s.recv(1024)
        return data
    finally:
        s.close()

File: tests/fixtures/real_world/ruby/state/socket_lifecycle.rb
require 'socket'

def connect_leak(host, port)
  s = TCPSocket.new(host, port)
  s.puts('hello')
  data = s.gets
  data
end

def connect_safe(host, port)
  s = TCPSocket.new(host, port)
  begin
    s.puts('hello')
    s.gets
  ensure
    s.close
  end
end
